<launch>
  <arg name="sim" default="true" />
  <param name="/use_sim_time" value="$(arg sim)" />
  <arg name="bag_filename" unless="$(arg sim)" /> <!--bag_namefile is not required if sim=false-->

  <group if="$(arg sim)">
    <param name="robot_description"
    command="$(find xacro)/xacro.py '$(find iai_boxy_description)/robots/boxy_description.urdf.xacro'" />
  </group>

    <node name="laser_remap" pkg="cartographer_ros"
    type="remap_laser_scan.py" respawn="false" output="screen"/>

  <node name="cartographer_node" pkg="cartographer_ros"
      type="cartographer_node" args="
          -configuration_directory
              $(find cartographer_ros)/configuration_files
          -configuration_basename boxy.lua"
      output="screen">
    <remap from="scan" to="/laser_scans" />
    <!--remap from="scan" to="/hokuyo_utm30_front/most_intense" /-->
    <!--remap from="scan" to="/base_scans" /-->
  </node>

 <!-- If it is using a BAG file (simulation) -->
  <group if="$(arg sim)">
    <node name="playbag" pkg="rosbag" type="play"
          args="--clock $(arg bag_filename)">
        <remap from="tf" to="tf_in" />
    </node>
    <node name="tf_remove_frames" pkg="cartographer_ros"
        type="tf_remove_frames.py">
      <remap from="tf_out" to="/tf" />
      <rosparam param="remove_frames">
        - map
        <!-- - odom-->
      </rosparam>
    </node>
  </group>

 <!-- If you're using the robot (not simulation) -->
  <!--group unless="$(arg sim)">
    <node name="tf_remove_frames" pkg="cartographer_ros"
        type="tf_remove_frames.py">
      <remap from="tf_out" to="/tf" />
      <rosparam param="remove_frames">
        - map
      </rosparam>
    </node>
  </group-->


  <include file="$(find cartographer_ros)/launch/boxy_rviz.launch"/>

</launch>
